src/simplay/src/simplay/visualization.py

```diff
@@ -29,11 +35,85 @@ def set_area(
         :param y: The y coordinate (row) of the top left corner of the area.
         :param color: The color of the area.
         """
+        if not isinstance(id, str):
+            raise ValueError("id must be a string")
+        if not isinstance(name, str):
+            raise ValueError("name must be a string")
+        if not isinstance(height, int):
+            raise ValueError("height must be an integer")
+        if not isinstance(width, int):
+            raise ValueError("width must be an integer")
+        if not isinstance(x, int):
+            raise ValueError("x must be an integer")
+        if not isinstance(y, int):
+            raise ValueError("y must be an integer")
+        if not isinstance(color, int):
+            raise ValueError("color must be an integer")
+        if x < 0 or x >= self.cols:
+            raise ValueError("x must be between 0 and cols")
+        if y < 0 or y >= self.rows:
+            raise ValueError("y must be between 0 and rows")
+        if x + width > self.cols:
+            raise ValueError(
+                "x added to width must be less than or equal to cols")
+        if y + height > self.rows:
+            raise ValueError(
+                "y added to height must be less than or equal to rows")
+        if height == 0 or width == 0:
+            raise ValueError("height and width must be greater than 0")
+        if self.check_overlap(id, height, width, x, y):
+            raise ValueError("area overlaps with another area")
+        if not self.check_areaid_unique(id):
+            raise ValueError("id must be unique")
         self.areas.append(
             {
                 "id": id,
                 "name": name,
                 "color": color,
-                "gridDefinition": {"height": height, "width": width, "x": x, "y": y},
+                "gridDefinition": {
+                    "height": height,
+                    "width": width,
+                    "x": x,
+                    "y": y
+                },
             }
         )
+
+    def check_areaid_unique(self, id: str):
+        """
+        Check if an area id is unique.
+
+        :param id: The id of the area.
+        :return: True if the id is unique, False otherwise.
+        """
+        for area in self.areas:
+            if area["id"] == id:
+                return False
+        return True
+
+    def check_overlap(self, id: str, height: int, width: int, x: int, y: int):
```

Comment:
Who calls that? 
If we really need this it can stay, but looping through all areas everytime this is called is kind of expensive

Reply:
this is needed to make sure, that no two areas have the same id, otherwise we risk having broken events when refering to areas in the future.